src/app/auth/reset-password/reset-password.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { AuthActions } from '../../common/auth';
+import { getAuthState } from '../../common/store/reducers';
+
+@Component({
+  selector: 'app-reset-password',
+  templateUrl: './reset-password.component.html',
+  styleUrls: ['./reset-password.component.scss']
+})
+export class ResetPasswordComponent {
+  public username = '';
+  public errorMessage = '';
+  public successMessage = '';
+
+  constructor(private store: Store<any>, private authActions: AuthActions) { }
+
+  reset_password() {
+    this.store.dispatch(this.authActions.resetPassword(this.username));
+    this.store.select(getAuthState).subscribe(({ isLoggedIn, inProgress, resetPassword, error }) => {
```

Comment:
Subscribing should be moved out of reset_password as each call creates a new subscription.